edge-util/src/Microsoft.Azure.Devices.Edge.Util/VerifyTwinSignature.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Security.Cryptography.X509Certificates;
+
+    using Org.Webpki.JsonCanonicalizer;
+    public class VerifyTwinSignature
+    {
+        public static bool VerifyModuleTwinSignature(string desiredProperties, string header, byte[] signatureBytes, X509Certificate2 signerCert, string algorithmScheme, HashAlgorithmName hashAlgorithm)
+        {
+            try
+            {
+                JsonCanonicalizer canonicalizerDesiredProperties = new JsonCanonicalizer(desiredProperties);
+                byte[] canonicalizedDesiredProperties = canonicalizerDesiredProperties.GetEncodedUTF8();
+
+                JsonCanonicalizer canonicalizerHeader = new JsonCanonicalizer(header);
+                byte[] canonicalizedHeader = canonicalizerHeader.GetEncodedUTF8();
+
+                byte[] canonicalizedCombinedDesiredProperties = new byte[canonicalizedHeader.Length + canonicalizedDesiredProperties.Length];
+
+                Array.Copy(canonicalizedHeader, 0, canonicalizedCombinedDesiredProperties, 0, canonicalizedHeader.Length);
+                Array.Copy(canonicalizedDesiredProperties, 0, canonicalizedCombinedDesiredProperties, canonicalizedHeader.Length, canonicalizedDesiredProperties.Length);
+
+                if (algorithmScheme == "ES")
+                {
+                    ECDsa eCDsa = signerCert.GetECDsaPublicKey();
+                    return eCDsa.VerifyData(canonicalizedCombinedDesiredProperties, signatureBytes, hashAlgorithm);
+                }
+                else if (algorithmScheme == "RS")
+                {
+                    RSA rsa = signerCert.GetRSAPublicKey();
+                    RSASignaturePadding rsaSignaturePadding = RSASignaturePadding.Pkcs1;
+                    return rsa.VerifyData(canonicalizedCombinedDesiredProperties, signatureBytes, hashAlgorithm, rsaSignaturePadding);
+                }
+                else
+                {
+                    throw new Exception("DSA Algorithm Type not supported");
+                }
+            }
+            catch (Exception ex)
```

Comment:
Here you can just catch and throw without cloning to a new exception instance.

Reply:
I want to catch the specific exception if DSA algorithm is not specified properly and DSA not supported and there could be other exceptions which would be array overflow or misuse. Thats why the new exception instance